autoemulate/compare.py

```diff
@@ -368,6 +378,74 @@ def load_model(self, path=None):
 
         return serialiser._load_model(path)
 
+    def print_settings(self):
+        if not self.is_set_up:
+            raise RuntimeError("Must run setup() before print_settings()")
+            return
+
+        models = "\n- " + "\n- ".join(
+            [
+                x[1].__class__.__name__
+                for pipeline in self.models
+                for x in pipeline.steps
+                if x[0] == "model"
+            ]
+        )
+        metrics = "\n- " + "\n- ".join([metric.__name__ for metric in self.metrics])
+
+        settings = pd.DataFrame(
+            [
+                str(self.X.shape),
+                str(self.y.shape),
+                str(self.train_idxs.shape),
+                str(self.test_idxs.shape),
+                str(self.param_search),
+                str(self.search_type),
+                str(self.param_search_iters),
+                str(self.scale),
+                str(
+                    self.scaler.__class__.__name__ if self.scaler is not None else None
+                ),
+                str(self.reduce_dim),
+                str(
+                    self.dim_reducer.__class__.__name__
+                    if self.dim_reducer is not None
+                    else None
+                ),
+                str(self.cv.__class__.__name__ if self.cv is not None else "None"),
+                str(self.folds),
+                str(self.n_jobs),
+            ],
+            index=[
+                "Simulation input shape (X)",
+                "Simulation output shape (y)",
+                "Training dataset shape (train_idxs)",
+                "Test dataset shape (test_idxs)",
```

Comment:
Just thinking, so these are just the indices, I think we can just say:

```
"# training set samples"
"# test set samples"

```
And do [0] above to get an `int` instead of a `tuple`.

Reply:
Addresed in https://github.com/alan-turing-institute/autoemulate/pull/195/commits/75d5ae61506dd76c7dd2ad3157c3cdc05c4e19e9